components/org.wso2.carbon.identity.scim2.common/src/test/java/org/wso2/carbon/identity/scim2/common/listener/SCIMUserOperationListenerTest.java

```diff
@@ -0,0 +1,358 @@
+import org.testng.annotations.DataProvider;
+import org.testng.annotations.ObjectFactory;
+import org.testng.annotations.Test;
+import org.wso2.carbon.identity.core.util.IdentityCoreConstants;
+import org.wso2.carbon.identity.scim2.common.DAO.GroupDAO;
+import org.wso2.carbon.identity.scim2.common.group.SCIMGroupHandler;
+import org.wso2.carbon.identity.scim2.common.utils.SCIMCommonUtils;
+import org.wso2.carbon.user.api.Permission;
+import org.wso2.carbon.user.api.RealmConfiguration;
+import org.wso2.carbon.user.core.UserStoreManager;
+import org.wso2.carbon.user.core.util.UserCoreUtil;
+import org.wso2.charon3.core.schema.SCIMConstants;
+
+import java.util.HashMap;
+import java.util.Map;
+import java.util.UUID;
+
+import static org.mockito.Matchers.any;
+import static org.mockito.Matchers.anyInt;
+import static org.mockito.Matchers.anyObject;
+import static org.mockito.Matchers.anyString;
+import static org.mockito.Matchers.eq;
+import static org.powermock.api.mockito.PowerMockito.mockStatic;
+import static org.powermock.api.mockito.PowerMockito.spy;
+import static org.powermock.api.mockito.PowerMockito.when;
+import static org.powermock.api.mockito.PowerMockito.whenNew;
+import static org.testng.Assert.assertEquals;
+import static org.testng.Assert.assertNotNull;
+import static org.testng.Assert.assertTrue;
+
+@PrepareForTest({UserCoreUtil.class, SCIMGroupHandler.class, SCIMCommonUtils.class})
+public class SCIMUserOperationListenerTest extends PowerMockTestCase {
+
+    private String userName = "testUser";
+    private Object credential = new Object();
+    private String roleName = "testRole";
+    private String[] roleList = new String[]{"testRole1, testRole2"};
+    private String[] userList = new String[]{"testUser1, testUser2"};
+    private Permission[] permissions = new Permission[0];
+    private Map<String, String> claims = new HashMap<>();
+    private String profile = "testProfile";
+    private boolean isAuthenticated = true;
+    SCIMUserOperationListener scimUserOperationListener;
+
+    @Mock
+    UserStoreManager userStoreManager;
+
+    @Mock
+    GroupDAO groupDAO;
+
+    @ObjectFactory
+    public IObjectFactory getObjectFactory() {
+        return new org.powermock.modules.testng.PowerMockObjectFactory();
+    }
+
+    @BeforeMethod
+    public void setUp() throws Exception {
+        scimUserOperationListener = spy(new SCIMUserOperationListener());
+        mockStatic(UserCoreUtil.class);
+        mockStatic(SCIMCommonUtils.class);
+    }
+
+    @AfterMethod
+    public void tearDown() throws Exception {
+
+    }
+
+    @DataProvider(name = "testGetExecutionOrderIdData")
+    public Object[][] testGetExecutionOrderIdData() {
+
+        return new Object[][]{
+                {10, 10},
+                {IdentityCoreConstants.EVENT_LISTENER_ORDER_ID, 90}
+        };
+    }
+
+    @Test(dataProvider = "testGetExecutionOrderIdData")
+    public void testGetExecutionOrderId(int orderID, int expectedResult) throws Exception {
```

Comment:
add a blank line

Reply:
As for our new convention, we have decided to remove the newline after the method signature.